src/test/java/unit/BqDatasetNumTables.java

```diff
@@ -153,5 +176,167 @@ void numTablesForReferencedWithNoAccess() throws IOException {
     // the external dataset created in the beforeall method should have 1 table in it, but the
     // sharee user doesn't have read access to the dataset so they can't know that
     assertNull(describeDataset.numTables, "referenced dataset with no access contains NULL tables");
+
+    // Clean up.
+    workspaceCreator.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+    TestCommand.runCommandExpectSuccess("resource", "delete", "--name=" + name, "--quiet");
+    TestCommand.runCommandExpectSuccess("workspace", "delete", "--quiet");
```

Comment:
these tests that create/delete a workspace in each method will leak workspaces if they fail. suggest looking at other classes that create multiple workspaces (e.g. `WorkspaceOverride`). I'm also not sure you need to create a new workspace every time. if the `shareeUser` is the same and has the same role every time, then you should be able to just reuse the same workspace and create resources with names that are unique to each test method.

Reply:
in some test, the sharee user needs to have write access to perform add-ref and update. but if i moved it to another class, this won't be an issue.